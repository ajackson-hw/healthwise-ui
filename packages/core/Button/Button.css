.button {
  padding: 0.75rem 1rem;
  border: 1px solid var(--color-primary);
  display: inline-flex;
  align-items: center;
  background-color: var(--color-primary);
  color: var(--color-text-on-primary);
  font-size: 1em;
  font-style: normal;
  text-decoration: none;
  line-height: 1;
  cursor: pointer;
}

.button:focus {
  outline: var(--focus-indicator);
  outline-offset: var(--focus-indicator-offset);
}

.button svg {
  width: 1em;
  height: 1em;
  fill: var(--color-text-on-primary);
}

.child:not(:last-child) {
  margin-right: 0.5rem;
}

.primary-light {
  border-color: var(--color-primary-light);
  background-color: var(--color-primary-light);
  color: var(--color-text-on-primary-light);
}

.primary-light svg {
  fill: var(--color-text-on-primary-light);
}

.primary-dark {
  border-color: var(--color-primary-dark);
  background-color: var(--color-primary-dark);
  color: var(--color-text-on-primary-dark);
}

.primary-dark svg {
  fill: var(--color-text-on-primary-dark);
}

.primary-darker {
  border-color: var(--color-primary-darker);
  background-color: var(--color-primary-darker);
  color: var(--color-text-on-primary-darker);
}

.primary-darker svg {
  fill: var(--color-text-on-primary-darker);
}

.accent {
  border-color: var(--color-accent);
  background-color: var(--color-accent);
  color: var(--color-text-on-accent);
}

.accent svg {
  fill: var(--color-text-on-accent);
}

.accent-dark {
  border-color: var(--color-accent-dark);
  background-color: var(--color-accent-dark);
  color: var(--color-text-on-accent-dark);
}

.accent-dark svg {
  fill: var(--color-text-on-accent-dark);
}

.neutral {
  border-color: var(--color-neutral);
  background-color: var(--color-neutral);
  color: var(--color-text-on-neutral);
}

.neutral svg {
  fill: var(--color-text-on-neutral);
}

.neutral-light {
  border-color: var(--color-neutral-light);
  background-color: var(--color-neutral-light);
  color: var(--color-text-on-neutral-light);
}

.neutral-light svg {
  fill: var(--color-text-on-neutral-light);
}

.neutral-dark {
  border-color: var(--color-neutral-dark);
  background-color: var(--color-neutral-dark);
  color: var(--color-text-on-neutral-dark);
}

.neutral-dark svg {
  fill: var(--color-text-on-neutral-dark);
}

.rounded {
  border-radius: 4px;
}

.flat {
  border: none;
  padding: 0.75rem;
  background-color: transparent;
  color: var(--color-primary);
}

.flat svg {
  fill: var(--color-primary);
}

.raised {
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1);
}

.raised:not(.disabled):hover,
.raised:not(.disabled):active {
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.4);
}

.outlined {
  background-color: transparent;
  color: var(--color-primary);
}

.outlined svg {
  fill: var(--color-primary);
}

.primary-light.outlined,
.primary-light.flat {
  color: var(--color-primary-light);
}

.primary-light.outlined svg,
.primary-light.flat svg {
  fill: var(--color-primary-light);
}

.primary-dark.outlined,
.primary-dark.flat {
  color: var(--color-primary-dark);
}

.primary-dark.outlined svg,
.primary-dark.flat svg {
  fill: var(--color-primary-dark);
}

.primary-darker.outlined,
.primary-darker.flat {
  color: var(--color-primary-darker);
}

.primary-darker.outlined svg,
.primary-darker.flat svg {
  fill: var(--color-primary-darker);
}

.accent.outlined,
.accent.flat {
  color: var(--color-accent);
}

.accent.outlined svg,
.accent.flat svg {
  fill: var(--color-accent);
}

.accent-dark.outlined,
.accent-dark.flat {
  color: var(--color-accent-dark);
}

.accent-dark.outlined svg,
.accent-dark.flat svg {
  fill: var(--color-accent-dark);
}

.neutral.outlined,
.neutral.flat {
  color: var(--color-neutral);
}

.neutral.outlined svg,
.neutral.flat svg {
  fill: var(--color-neutral);
}

.neutral-light.outlined,
.neutral-light.flat {
  color: var(--color-neutral-light);
}

.neutral-light.outlined svg,
.neutral-light.flat svg {
  fill: var(--color-neutral-light);
}

.neutral-dark.outlined,
.neutral-dark.flat {
  color: var(--color-neutral-dark);
}

.neutral-dark.outlined svg,
.neutral-dark.flat svg {
  fill: var(--color-neutral-dark);
}

.primary-light.outlined:focus,
.primary-light.flat:focus,
.neutral-light.outlined:focus,
.neutral-light.flat:focus {
  outline: var(--focus-indicator-contrast);
}

.disabled,
.disabled:hover,
.disabled:active {
  opacity: 0.35;
  pointer-events: none;
}
