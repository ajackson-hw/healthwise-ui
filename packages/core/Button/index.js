import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './Button.css'

class Button extends Component {
  render() {
    const {
      className,
      disabled,
      flat,
      href,
      outlined,
      raised,
      render,
      rounded,
      theme,
      ...otherProps
    } = this.props

    const buttonClass = classNames(
      'hw-button',
      styles.button,
      styles[theme],
      { [styles.rounded]: rounded },
      { [styles.flat]: flat },
      { [styles.raised]: raised },
      { [styles.outlined]: outlined },
      { [styles.disabled]: disabled },
      className
    )

    const ariaProps = href || otherProps.to ? { role: 'button', 'aria-disabled': disabled } : {}

    // Wrap all children in a div to allow setting margin on text nodes
    const children = React.Children.map(this.props.children, (child, index) => (
      <div key={index} className={styles.child}>
        {child}
      </div>
    ))

    const props = {
      className: buttonClass,
      disabled,
      href,
      ...ariaProps,
      ...otherProps,
    }

    if (render) {
      return render({ ...props, children })
    } else if (href) {
      return <a {...props}>{children}</a>
    } else {
      return <button {...props}>{children}</button>
    }
  }
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  render: PropTypes.func,
  disabled: PropTypes.bool,
  flat: PropTypes.bool,
  href: PropTypes.string,
  id: PropTypes.string,
  outlined: PropTypes.bool,
  raised: PropTypes.bool,
  rounded: PropTypes.bool,
  theme: PropTypes.oneOf([
    'primary',
    'primary-light',
    'primary-dark',
    'primary-darker',
    'accent',
    'accent-dark',
    'neutral',
    'neutral-light',
    'neutral-dark',
  ]),
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
}

Button.defaultProps = {
  theme: 'primary',
}

export default Button
