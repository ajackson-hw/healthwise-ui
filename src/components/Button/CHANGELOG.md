# Change Log

## 4.0.0 - August 14, 2018

- **BREAKING**: Changed styling of `flat` style buttons
- Added `render` prop to control markup of rendered button
- Added `outlined` prop
- Added `primary-light`, `primary-dark`, `primary-darker`, `accent-dark`, `neutral-light`, and `neutral-dark` theme variants

## 3.0.0 - July 9, 2018

- **BREAKING**: Updated `@healthwise/theme` to `v4`, and removed deprecated theme variables

## 2.0.3 - May 24, 2018

- Version bump. Testing publishing strategies.

## 2.0.2 - May 24, 2018

- Updated package-lock files
- Updated snapshots for newer version of Jest in `react-app-config`

## 2.0.1 - May 8, 2018

- Added `package-lock.json` files
- Ensure consistent package dependencies
- Update React peer dependency to `>=15.4.2` to remove warnings on React 16

## 2.0.0 - April 27, 2018

- Updated styling
- Updated theme variables introduced in `@healthwise/theme v3`
- Added `theme` prop which supports `primary`, `accent`, and `neutral` themes

## 1.1.1 - December 27, 2017

- Added `@healthwise/theme` to peer dependencies

## 1.1.0 - Initial release
