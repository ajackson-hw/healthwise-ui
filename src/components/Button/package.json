{
  "name": "@healthwise/button",
  "version": "4.0.0",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - Button component",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "classnames": "^2.2.5",
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "@healthwise/theme": "^4.0.0",
    "react": ">=15.4.2"
  }
}
