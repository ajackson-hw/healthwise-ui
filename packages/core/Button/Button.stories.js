import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import Theme from '../Theme'
import { ProfileIcon } from '../Icon'
import Button from './index'

const theme = Theme()

const styles = {
  row: {
    display: 'flex',
    flexFlow: 'row wrap',
  },
  header: {
    paddingLeft: '16px',
  },
  container: {
    padding: '16px',
  },
  container_light: {
    padding: '16px',
    background: theme['color-background-light'],
  },
  container_contrast: {
    padding: '16px',
    background: theme['color-background-contrast-dark'],
  },
}

const onClickAction = action('onClick')
const handleClick = event => {
  event.preventDefault()
  onClickAction(event)
}

storiesOf('Form Controls/Button', module)
  .add('with defaults', () => (
    <div style={styles.row}>
      <div style={styles.container}>
        <Button>Default</Button>
      </div>
      <div style={styles.container}>
        <Button raised>Raised</Button>
      </div>
      <div style={styles.container}>
        <Button rounded>Rounded</Button>
      </div>
      <div style={styles.container}>
        <Button flat>Flat</Button>
      </div>
      <div style={styles.container}>
        <Button outlined>Outlined</Button>
      </div>
      <div style={styles.container}>
        <Button rounded outlined>
          Rounded Outlined
        </Button>
      </div>
    </div>
  ), {
  info: `Demonstrates default rendering of a Button`
})
  .add('with themes', () => (
    <div>
      <h2 style={styles.header}>Primary (default)</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button>
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container_light}>
          <Button flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container_light}>
          <Button rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Primary Light</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="primary-light">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-light" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-light" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container_contrast}>
          <Button theme="primary-light" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container_contrast}>
          <Button theme="primary-light" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Primary Dark</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="primary-dark">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-dark" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-dark" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-dark" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-dark" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Primary Darker</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="primary-darker">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-darker" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-darker" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-darker" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="primary-darker" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Accent</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="accent">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="accent" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="accent" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container_light}>
          <Button theme="accent" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container_light}>
          <Button theme="accent" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Accent Dark</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="accent-dark">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="accent-dark" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="accent-dark" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="accent-dark" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="accent-dark" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Neutral</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="neutral">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container_light}>
          <Button theme="neutral" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container_light}>
          <Button theme="neutral" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Neutral Light</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="neutral-light">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral-light" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral-light" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container_contrast}>
          <Button theme="neutral-light" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container_contrast}>
          <Button theme="neutral-light" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Neutral Dark</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button theme="neutral-dark">
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral-dark" raised>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral-dark" rounded>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral-dark" flat>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container}>
          <Button theme="neutral-dark" rounded outlined>
            <ProfileIcon /> Outlined
          </Button>
        </div>
      </div>
    </div>
  ), {
  info: `Demonstrates available themes for a Button`
})
  .add('with icon only', () => (
    <div style={styles.row}>
      <div style={styles.container}>
        <Button>
          <ProfileIcon />
        </Button>
      </div>
      <div style={styles.container}>
        <Button raised>
          <ProfileIcon />
        </Button>
      </div>
      <div style={styles.container}>
        <Button rounded>
          <ProfileIcon />
        </Button>
      </div>
      <div style={styles.container}>
        <Button flat>
          <ProfileIcon />
        </Button>
      </div>
      <div style={styles.container}>
        <Button outlined>
          <ProfileIcon />
        </Button>
      </div>
      <div style={styles.container}>
        <Button rounded outlined>
          <ProfileIcon />
        </Button>
      </div>
    </div>
  ), {
  info: `Demonstrates a button with only an icon child`
})
  .add('with links', () => (
    <div style={styles.row}>
      <div style={styles.container}>
        <Button href="#" onClick={handleClick}>
          <ProfileIcon /> Default
        </Button>
      </div>
      <div style={styles.container}>
        <Button href="#" onClick={handleClick} raised>
          <ProfileIcon /> Raised
        </Button>
      </div>
      <div style={styles.container}>
        <Button href="#" onClick={handleClick} rounded>
          <ProfileIcon /> Rounded
        </Button>
      </div>
      <div style={styles.container}>
        <Button href="#" onClick={handleClick} flat>
          <ProfileIcon /> Flat
        </Button>
      </div>
      <div style={styles.container}>
        <Button href="#" onClick={handleClick} outlined>
          <ProfileIcon /> Outlined
        </Button>
      </div>
      <div style={styles.container}>
        <Button href="#" onClick={handleClick} rounded outlined>
          <ProfileIcon /> Rounded Outlined
        </Button>
      </div>
    </div>
  ), {
  info: `Demonstrates a button rendered as a link`
})
  .add('disabled', () => (
    <div>
      <h2 style={styles.header}>Regular Buttons</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button disabled onClick={onClickAction}>
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button raised disabled onClick={onClickAction}>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button rounded disabled onClick={onClickAction}>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container}>
          <Button flat disabled onClick={onClickAction}>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container}>
          <Button outlined disabled onClick={onClickAction}>
            <ProfileIcon /> Outlined
          </Button>
        </div>
        <div style={styles.container}>
          <Button rounded outlined disabled onClick={onClickAction}>
            <ProfileIcon /> Rounded Outlined
          </Button>
        </div>
      </div>
      <h2 style={styles.header}>Link Buttons</h2>
      <div style={styles.row}>
        <div style={styles.container}>
          <Button href="#" disabled onClick={handleClick}>
            <ProfileIcon /> Default
          </Button>
        </div>
        <div style={styles.container}>
          <Button href="#" raised disabled onClick={handleClick}>
            <ProfileIcon /> Raised
          </Button>
        </div>
        <div style={styles.container}>
          <Button href="#" rounded disabled onClick={handleClick}>
            <ProfileIcon /> Rounded
          </Button>
        </div>
        <div style={styles.container}>
          <Button href="#" flat disabled onClick={handleClick}>
            <ProfileIcon /> Flat
          </Button>
        </div>
        <div style={styles.container}>
          <Button href="#" outlined disabled onClick={handleClick}>
            <ProfileIcon /> Outlined
          </Button>
        </div>
        <div style={styles.container}>
          <Button href="#" rounded outlined disabled onClick={handleClick}>
            <ProfileIcon /> Rounded Outlined
          </Button>
        </div>
      </div>
    </div>
  ), {
  info: `Demonstrates a disabled Button`
})
