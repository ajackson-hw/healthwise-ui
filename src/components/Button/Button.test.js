/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Button from './index'

describe('<Button />', () => {
  it('renders correctly with defaults', () => {
    const component = renderer.create(<Button>Works</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with id and added classname', () => {
    const component = renderer.create(
      <Button id="idTest" className="classNameTest">
        Works
      </Button>
    )
    expect(component).toMatchSnapshot()
  })

  it('renders correctly as submit', () => {
    const component = renderer.create(<Button type="submit">Works</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly as type button', () => {
    const component = renderer.create(<Button type="button">Works</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly as type reset', () => {
    const component = renderer.create(<Button type="reset">Works</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with text child', () => {
    const component = renderer.create(<Button>Text</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with complex children', () => {
    const component = renderer.create(
      <Button>
        <span>An element</span> Text
      </Button>
    )
    expect(component).toMatchSnapshot()
  })

  it('renders correctly as a link button', () => {
    const component = renderer.create(<Button href="https://www.google.com">Child</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders rounded style correcty', () => {
    const component = renderer.create(<Button rounded>Child</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders flat style correcty', () => {
    const component = renderer.create(<Button flat>Child</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders raised style correcty', () => {
    const component = renderer.create(<Button raised>Child</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders primary theme correctly', () => {
    const component = renderer.create(<Button theme="primary">Primary button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders accemt theme correctly', () => {
    const component = renderer.create(<Button theme="accent">Accent button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders neutral theme correctly', () => {
    const component = renderer.create(<Button theme="neutral">Neutral button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('accepts additional props', () => {
    const component = renderer.create(<Button aria-hidden="true">Child</Button>)
    expect(component).toMatchSnapshot()
  })
})
