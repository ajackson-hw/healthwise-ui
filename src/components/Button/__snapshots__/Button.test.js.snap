// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> accepts additional props 1`] = `
<button
  aria-hidden="true"
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Child
  </div>
</button>
`;

exports[`<Button /> renders accemt theme correctly 1`] = `
<button
  className="hw-button button accent"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Accent button
  </div>
</button>
`;

exports[`<Button /> renders correctly as a link button 1`] = `
<a
  aria-disabled={undefined}
  className="hw-button button primary"
  disabled={undefined}
  href="https://www.google.com"
  role="button"
>
  <div
    className="child"
  >
    Child
  </div>
</a>
`;

exports[`<Button /> renders correctly as submit 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
  type="submit"
>
  <div
    className="child"
  >
    Works
  </div>
</button>
`;

exports[`<Button /> renders correctly as type button 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
  type="button"
>
  <div
    className="child"
  >
    Works
  </div>
</button>
`;

exports[`<Button /> renders correctly as type reset 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
  type="reset"
>
  <div
    className="child"
  >
    Works
  </div>
</button>
`;

exports[`<Button /> renders correctly with complex children 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    <span>
      An element
    </span>
  </div>
  <div
    className="child"
  >
     Text
  </div>
</button>
`;

exports[`<Button /> renders correctly with defaults 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Works
  </div>
</button>
`;

exports[`<Button /> renders correctly with id and added classname 1`] = `
<button
  className="hw-button button primary classNameTest"
  disabled={undefined}
  href={undefined}
  id="idTest"
>
  <div
    className="child"
  >
    Works
  </div>
</button>
`;

exports[`<Button /> renders correctly with text child 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Text
  </div>
</button>
`;

exports[`<Button /> renders flat style correcty 1`] = `
<button
  className="hw-button button primary flat"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Child
  </div>
</button>
`;

exports[`<Button /> renders neutral theme correctly 1`] = `
<button
  className="hw-button button neutral"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Neutral button
  </div>
</button>
`;

exports[`<Button /> renders primary theme correctly 1`] = `
<button
  className="hw-button button primary"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Primary button
  </div>
</button>
`;

exports[`<Button /> renders raised style correcty 1`] = `
<button
  className="hw-button button primary raised"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Child
  </div>
</button>
`;

exports[`<Button /> renders rounded style correcty 1`] = `
<button
  className="hw-button button primary rounded"
  disabled={undefined}
  href={undefined}
>
  <div
    className="child"
  >
    Child
  </div>
</button>
`;
